sdk/dotnet/Pulumi/Log.cs

```diff
@@ -31,15 +31,17 @@ public static void Warn(string message, Resource? resource = null, int? streamId
             => Deployment.InternalInstance.Logger.WarnAsync(message, resource, streamId, ephemeral);
 
         /// <summary>
-        /// Logs a fatal error to indicate that the tool should stop processing resource
-        /// operations immediately.
+        /// Logs a fatal condition. Calling Error does not by itself
+        /// stop processing resource operations. To stop programs
+        /// should raise unhandled exceptions after calling Error.
```

Comment:
Nit: something like "Raise an unhandled exception after calling Error to stop the program" seems more natural to me. Or "You should raise...".